packages/@aws-cdk/aws-ssm/lib/parameter.ts

```diff
@@ -302,8 +365,37 @@ export interface StringParameterAttributes extends CommonStringParameterAttribut
    * The type of the string parameter
    *
    * @default ParameterType.STRING
+   * @deprecated - use valueType instead
    */
   readonly type?: ParameterType;
+
+  /**
+   * The type of the string parameter value
+   *
+   * @default ParameterValueType.STRING
+   */
+  readonly valueType?: ParameterValueType;
+}
```

Comment:
The changes look fine, the meta-meta aspect for the names is making me do double-taken when reviewing them.

Could some of the other names you considered make it easier to read, even if it is more verbose?

Reply:
hmm I'm not sure, I was having a hard time coming up with an alternative for `type` ha. Any suggestions?

It's definitely confusing because we are modeling two different types of SSM parameters in the same construct, which each have the concept of `type`.